contracts/dao/proposal/cwd-proposal-multiple/src/testing/tests.rs

```diff
@@ -2128,3 +2132,83 @@ fn test_close_failed_proposal() {
     // not reverted
     assert_eq!(updated.proposal.status, Status::Passed);
 }
+
+#[test]
+fn test_reply_proposal_mock() {
+    use crate::contract::reply;
+    use crate::state::PROPOSALS;
+
+    let mut deps = mock_dependencies();
+    let env = mock_env();
+
+    let m_proposal_id = mask_proposal_execution_proposal_id(1);
+    PROPOSALS
+        .save(
+            deps.as_mut().storage,
+            1,
+            &MultipleChoiceProposal {
+                title: "A simple text proposal".to_string(),
+                description: "This is a simple text proposal".to_string(),
+                proposer: Addr::unchecked(CREATOR_ADDR),
+                start_height: env.block.height,
+                expiration: Duration::Height(6).after(&env.block),
+                min_voting_period: None,
+                allow_revoting: false,
+                total_power: Uint128::new(100_000_000),
+                status: Status::Open,
+                votes: MultipleChoiceVotes {
+                    vote_weights: vec![Uint128::zero(); 3],
+                },
+                choices: vec![],
+                voting_strategy: VotingStrategy::SingleChoice {
+                    quorum: PercentageThreshold::Majority {},
+                },
+            },
+        )
+        .unwrap();
+
+    // PROPOSALS
+    let reply_msg = Reply {
+        id: m_proposal_id,
+        result: SubMsgResult::Err("error".to_string()),
+    };
+    let res = reply(deps.as_mut(), env.clone(), reply_msg).unwrap();
+    assert_eq!(
+        res.attributes[0],
+        Attribute {
+            key: "proposal_execution_failed".to_string(),
+            value: 1.to_string()
+        }
+    );
+
+    let prop = PROPOSALS.load(deps.as_mut().storage, 1).unwrap();
+    assert_eq!(prop.status, Status::ExecutionFailed);
+
+    // reply writes the failed proposal error
+    let query_res = query_proposal_failed_execution_error(deps.as_ref(), 1).unwrap();
+    let query_errs: FailedProposalErrors = from_binary(&query_res).unwrap();
+    assert_eq!(query_errs.errors.len(), 1);
+    let error = query_errs.errors.first().unwrap();
+    assert_eq!(error.error, "error".to_string());
+    assert_eq!(error.height, env.block.height);
+
+    // reply second time appends new error
+    let env2 = {
+        let mut e = env;
+        e.block.height += 10;
+        e
+    };
+    let msg2 = Reply {
+        id: m_proposal_id,
+        result: SubMsgResult::Err("error2".to_string()),
+    };
+    let res_ok = reply(deps.as_mut(), env2.clone(), msg2).unwrap();
+    assert_eq!(0, res_ok.messages.len());
+
+    let query_res = query_proposal_failed_execution_error(deps.as_ref(), 1).unwrap();
+    let query_errs: FailedProposalErrors = from_binary(&query_res).unwrap();
+    assert_eq!(query_errs.errors.len(), 2);
+    let error2 = query_errs.errors.last().unwrap();
+    assert_eq!(error2.error, "error2".to_string());
+    assert_eq!(error2.height, env2.block.height);
```

Comment:
let's also check that the first recorded error info persists, and that the status is `Status::ExecutionFailed`, huh? here and in other test files